### PR TITLE
WASM: Configure `emcc` compilation+linking flags

### PR DESCRIPTION
**Pull request by evacchi**
- Configure some sensible defaults for `emcc` to compiler IR and link the Wasm+JS executable
- ~~Add Wasm to the list of Unix-like platforms~~
- Disable explicit linking of libunwind, gcc_s using `unless`
- Avoid invoking `mprotect()` during heap setup, since it is unsupported (prints a warning to stderr, causes the tests to fail)

Exceptions are still not supported (test TryCatch would fail, see also https://github.com/qbicc/qbicc/pull/1402)

### CODE REVIEW

File: runtime/api/src/main/java/org/qbicc/runtime/Build.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ public static boolean isVirtual() {
 
         @Fold
         public static boolean isUnix() {
-            return defined(__unix__) || isMacOs();
+            return defined(__unix__) || isMacOs() || isWasm();
```
Comment (dmlloyd) [inline runtime/api/src/main/java/org/qbicc/runtime/Build.java:133]: Is there an emscripten-specific constant we can test for instead? We might (for example) use WASM in a bare WASI environment which isn't very UNIX-ish; IOW "WASM" doesn't imply "UNIX".

Comment (evacchi) [reply runtime/api/src/main/java/org/qbicc/runtime/Build.java:133]: I think `__WASI__` would imply unix, but there should be `__emscripten__` or something like that. Will check

Comment (dmlloyd) [reply runtime/api/src/main/java/org/qbicc/runtime/Build.java:133]: The docs at https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md definitely imply a UNIX-like environment but I don't know if it is sufficiently UNIX-like to call it "UNIX".

I guess the definition of "UNIX" that we're actually using in qbicc is just "is there a `<unistd.h>`?" so maybe it's not actually a useful constant at all. For implementation we really only care if it's POSIX, Windows, or something else (and WASI would be something else *unless* we are using a POSIX API over the top of it, just as Windows could be considered POSIX *if* we used something like cygwin and had those POSIX header files and macros defined).

In conclusion, I guess it doesn't matter too much. We probably should remove `isUnix` altogether.

Comment (evacchi) [reply runtime/api/src/main/java/org/qbicc/runtime/Build.java:133]: I'll purge this commit from the PR since it's an orthogonal concern anyway
